test/xml/xmldocFileParserTest.ts

```diff
@@ -0,0 +1,262 @@
+import { TreeNode, TreeVisitor, visit } from "@atomist/tree-path";
+import * as assert from "assert";
+import { isXmlDocTreeNode, XmldocFileParser, XmlDocTreeNode } from "../../lib/xml/XmldocFileParser";
+import { springBootPom } from "../spring/generator/TestPoms";
+
+function positionVerifier(rawdoc: string): TreeVisitor {
+    return (n: TreeNode) => {
+        if (!!n.$parent) {
+            assert.strictEqual(n.$value, rawdoc.substr(n.$offset, n.$value.length),
+                "Invalid node offset");
+        }
+        return true;
+    };
+}
+
+describe("xmldocFileParser", () => {
+
+    describe("direct use", () => {
+
+        it("should parse empty to undefined", async () => {
+            const f = new InMemoryProjectFile("a.xml", "");
+            const parsed = await new XmldocFileParser().toAst(f);
+            assert.strictEqual(parsed, undefined);
+        });
+
+        it("should parse root only", async () => {
+            const f = new InMemoryProjectFile("a.xml", RootOnly);
+            const parsed = await new XmldocFileParser().toAst(f);
+            assert(!!parsed);
+            visit(parsed, positionVerifier(RootAndSingleChildElement));
+            assert.strictEqual(parsed.$name, "Root");
+            assert.strictEqual(parsed.$children.length, 0);
+        });
+
+        it("should parse root and single element only", async () => {
+            const f = new InMemoryProjectFile("a.xml", RootAndSingleChildElement);
+            const parsed = await new XmldocFileParser().toAst(f);
+            visit(parsed, positionVerifier(RootAndSingleChildElement));
+            assert(!!parsed);
+            assert.strictEqual(parsed.$name, "Root");
+            assert.strictEqual(parsed.$children.length, 1);
+            assert.strictEqual(parsed.$children[0].$name, "kid");
+            assert.strictEqual(parsed.$children[0].$children.length, 0);
+        });
+
+        it("should parse root and single element only, retaining offsets", async () => {
+            const f = new InMemoryProjectFile("a.xml", RootAndSingleChildElement);
+            const parsed = await new XmldocFileParser().toAst(f);
+            visit(parsed, positionVerifier(RootAndSingleChildElement));
+            assert(!!parsed);
+            assert.strictEqual(parsed.$name, "Root");
+            assert.strictEqual(parsed.$offset, RootAndSingleChildElement.indexOf("<Root"));
+            assert.strictEqual(parsed.$children.length, 1);
+            assert.strictEqual(parsed.$children[0].$name, "kid");
+            assert.strictEqual(parsed.$children[0].$children.length, 0);
+        });
+
+        it("should parse POM", async () => {
+            const f = new InMemoryProjectFile("pom.xml", springBootPom());
+            const parsed = await new XmldocFileParser().toAst(f);
+            // visit(parsed, positionVerifier(RootAndSingleChildElement));
+            assert(!!parsed);
+        });
+
+    });
+
+    describe("with path expressions", () => {
```

Comment:
Can you add an example where you add an author underneath the list of current authors (so the authors are still grouped together?)

Reply:
I've added a simple example of this. It really needs support for the `last()` functionality from Path. I'll raise an issue for that in `tree-path`